runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -47,8 +47,12 @@
     private LocalDateTime startTime;
     private long startAddress;
     private long endAddress;
+    @Getter
     private long numEntries = 0;
     private long numBytes = 0;
+    //SMR entries total size falling on EXACTLY on the write size limit set will create problems
+    //Making sure there is 2% room for the check to create two long check point entries instead
```

Comment:
I know that I mentioned `2%` overhead at some point but you shouldn't depend on this number. I would just opt for simplicity and not account for the overhead at all. Since the payload size can be tuned I feel like this is unnecessary (because its not accurate).

Reply:
@Maithem wouldnt a safety guard here better? Hypothetically speaking, there are chances of hitting the boundary even when the payload size is tunable and although, the chances of hitting this exception again after the payload is tuned and re-tried is slim, it still exists. Adding this measure would remove that slim chance altogether in my opinion.